### DISCUSSION TITLE
Replacing throw with JsValue.Undefined

### DISCUSSION THREAD

**Discussion by kotbrain** [sebastienros/jint#1431] [Q&A]
I am trying to make a `NodeList` implementation and found a issue. Following the [specification](https://www.w3.org/TR/DOM-Level-3-Core/core.html#ID-536297177):
`If index is greater than or equal to the number of nodes in the least, this return null.`

However, I get an exception in the `ReflectionAccessor.GetValue` method and code execution is interrupted.
`Index was out of range. Must be non-negative and less than the size of the collection.`

I replaced the throw with return `JsValue.Undefined` and that solved this issue. I could add checks to the `NodeList`, but it's not an elegant solution. Could you make a fixes to Jint or add customization of this behavior?

**Comment by lahma** [comment]
Can you show us your code?

**Comment by kotbrain** [comment]
I'm using the [AngleSharp class](https://github.com/AngleSharp/AngleSharp/blob/e98928f46966a3ee4cde076fb08129fceb0343eb/src/AngleSharp/Dom/Internal/NodeList.cs). 

Test JavaScript code:
```
var testElement = document.createElement('test');
console.log(testElement.childNodes[1]);
```

**Reply by lahma** [reply]
Well that implementation should probably check for valid index in indexer and return null if out of bounds. You could probably copy and modify your own version on your project.
